Improve custom block contribution doc (#1003)
flojoy-ai/studio: docs/src/content/docs/contribution/blocks/01-create-your-custom-block.md

## A basic example

In this tutorial we will create a block that divides two items element-wise (for the case of vector inputs, for instance). Although we could do this with the built-in `invert` and `multiply` blocks, we will do it from scratch.

### Creating the custom block `DIVIDE`

To start, we create a `custom-blocks` folder. You can name it as you wish. We will put our all custom blocks in this folder so that we can import them together.

> Now, we create a `DIVIDE` folder inside `custom-blocks` folder and inside `DIVIDE` folder create a new Python file with the same name as folder `DIVIDE.py`.

Reviewer: ```suggestion
Now, we create a `DIVIDE` folder inside `custom-blocks` folder. Inside the `DIVIDE` folder, create a new Python file with the same name as the folder, `DIVIDE.py`.
```

:::note

> Each block should have it's own folder. The name of the folder, file and block function should be the same and in uppercase. We use this convention to efficiently generate manifest from blocks and use it in Frontend to visualize correctly

Reviewer: ```suggestion
Each block should have its own folder. The name of the folder, file and block function should be the same and in uppercase. We use this convention to allow the blocks to be automatically loaded into Flojoy Studio.
```

Let's say we want to create a block to wrap the `train_test_split` function from `scikit-learn`. This block will have to return two different `DataContainers`.

> For that we start with creating a new folder inside `custom-blocks` directory called `TRAIN_TEST_SPLIT` and a Python file inside `TRAIN_TEST_SPLIT` folder with the same name `TRAIN_TEST_SPLIT.py`. Then we put following code in the file:

Reviewer: ```suggestion
We start by creating a new folder inside the `custom-blocks` directory called `TRAIN_TEST_SPLIT`, and a Python file inside the `TRAIN_TEST_SPLIT` folder with the same name: `TRAIN_TEST_SPLIT.py`. Then we put the following code in the file:
```

Looking at the parameters, we have one `DataContainer` input, called `default`. When we only have one input and we do not want to label it in the Front-End, we can name it `default`, which is a special name that Flojoy recognizes. This block also has a `test_size` parameter that has a default value of 0.2.

### Importing custom block in Flojoy

There are few steps to import your custom blocks to Flojoy:

1. First click on `Add block` button from top left of the Flojoy studio app. It will expand a side bar.

2. Click on `Custom` tab from the blocks sidebar. Here you'll find a button named `Import custom blocks`.

3. Click on `Import custom blocks` button. It'll open a file window.

> 4. Now head to your directory where you have created your custom block, in this case that folder is `custom-blocks` folder. So select the folder.

Reviewer: ```suggestion
4. Now head to your directory where you have created your custom block and select it. In this case that folder is `custom-blocks`.
```

Congratulations! you just imported your custom block to Flojoy. You should see your custom block in the sidebar. Click on the block to add it to flow chart.

:::note

> Every time you create a new block, Studio will hot reload to include it in the sidebar under the `Custom` tab. And any modification you do in your custom blocks Studio hot load will automatically sync your changes. No reload is required.

Reviewer: ```suggestion
Every time you create a new block, Studio will hot reload to include it in the sidebar under the `Custom` tab. Any changes to custom blocks are immediately reflected in Studio, no reload is required.
```
